tidb-binlog/pump_client/client.go

```diff
@@ -416,13 +418,21 @@ func (c *PumpsClient) addPump(pump *PumpStatus, updateSelector bool) {
 		c.Selector.SetPumps(copyPumps(c.Pumps.AvaliablePumps))
 	}
 
-	c.Pumps.Unlock()
+	c.Unlock()
+}
+
+// SetSelectStrategy sets the selector's strategy, strategy should be 'range' or 'hash' now.
+func (c *PumpsClient) SetSelectStrategy(strategy string) {
+	c.Lock()
+	c.Selector = NewSelector(strategy)
```

Comment:
will you check which is really supported before call SetSelectStrategy? like setting `score` must has some problem now.